core-tests/src/test/java/org/mule/runtime/core/context/DefaultMuleContextTestCase.java

```diff
@@ -108,6 +109,8 @@ public void callSystemExceptionHandlerWhenExceptionIsMessagingException() throws
         context.setExceptionListener(mockSystemExceptionHandler);
         context.handleException(mockMessagingException);
         verify(mockSystemExceptionHandler, VerificationModeFactory.times(1)).handleException(mockMessagingException,null);
+
+        context.dispose();
```

Comment:
shouldn't this context be a class field and have an after() which cleans it up if not null?